src/imgui/fileHandler.hpp

```diff
@@ -6,35 +6,55 @@
 #include <string>
 
 namespace cashoverflow::utils {
-struct FileHandler
+
+class IFileHandler {
+  public:
+    virtual std::string read() const = 0;
+    virtual unsigned int read(char *buffer, int size) = 0;
+    virtual std::string getFileName() const = 0;
+    virtual ~IFileHandler() = default;
+};
+class FileHandler: public IFileHandler
 {
-  explicit FileHandler(std::string fileName_) : fileName(std::move(fileName_))
-  {
-    fileStream.open(fileName.c_str(), std::ios_base::in);
-    if (!fileStream.is_open()) {
-      throw std::runtime_error("could not open the file.");
+  public:
+    explicit FileHandler(std::string fileName_) : fileName(std::move(fileName_))
+    {
+      fileStream.open(fileName.c_str(), std::ios_base::in);
+      if (!fileStream.is_open()) {
+        throw std::runtime_error("could not open the file.");
```

Comment:
I would rather have a private constructor + a static factory method that returns an expected, similar to what you did for the account. The problem with throwing in the constructor (besides all the problems with throwing) is if we have something like this:

```c++
struct A {
    FileHandler fh{"file.txt"};
};
```

Then we would have to use C++'s horrible syntax for solving this.